cognito-post-authentication/src/test/java/no/unit/nva/cognito/PostAuthenticationHandlerTest.java

```diff
@@ -36,31 +44,56 @@
     @BeforeEach
     public void init() {
         customerApi = mock(CustomerApi.class);
+        userApi = mock(UserApi.class);
         awsCognitoIdentityProvider = mock(AWSCognitoIdentityProvider.class);
-        handler = new PostAuthenticationHandler(customerApi, awsCognitoIdentityProvider);
+        handler = new PostAuthenticationHandler(new UserService(userApi, awsCognitoIdentityProvider), customerApi);
     }
 
     @Test
     public void handleRequestReturnsEventOnInput() {
         UUID customerId = UUID.randomUUID();
-        Map<String,Object> requestEvent = Map.of(
+        prepareMocksWithValidResponse(customerId);
```

Comment:
Here we should have tests like 

* "handler creates User with Publiser role for stuff member", 
* "handler creates User with User role for non-stuff member"
* "handler propagates User details for existing User"
* "handler throws error when User cannot be parsed" 
and so on...

In this way we will be able to document/understand/remember what the Handler is supposed to do in each case and which cases have been covered. They could be written as cucumber tests but Unit tests are also quite sufficient.